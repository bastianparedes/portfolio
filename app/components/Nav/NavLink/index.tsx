import React from 'react';

import styles from './styles.module.scss';

interface NavLinkProps {
  link: string;
  title: string;
  closeNav: () => void;
}

const NavLink = ({ link, title, closeNav }: NavLinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a className={styles.link__a} href={'#' + link} onClick={closeNav}>
        {title}
      </a>
    </li>
  );
};

export default NavLink;
