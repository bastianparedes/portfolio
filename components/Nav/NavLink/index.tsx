import React, { MouseEventHandler } from 'react'
import styles from './styles.module.css'

interface NavLinkProps {
  link: string
  title: string
  onClick: MouseEventHandler<HTMLAnchorElement>
}

const NavLink = (props: NavLinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a
        href={'#' + props.link}
        className={styles.link__a}
        onClick={props.onClick}
      >
        {props.title}
      </a>
    </li>
  )
}

export default NavLink
