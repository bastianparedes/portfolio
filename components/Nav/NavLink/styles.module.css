.link {
}

.link__a {
  color: rgb(255, 255, 255);
  font-size: inherit;
  text-decoration: none;
}

.link__a:hover {
  color: rgb(103, 255, 124);
}

@media (max-device-width: 700px) {
  .link {
    width: 100%;
    height: 4rem;
    display: flex;
    align-items: center;
    justify-content: center;
  }
}
