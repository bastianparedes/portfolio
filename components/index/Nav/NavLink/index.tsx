import React from 'react';

import styles from './styles.module.scss';

interface NavLinkProps {
  link: string;
  title: string;
  closeNav: () => void;
}

const NavLink = ({ link, title, closeNav }: NavLinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a href={'#' + link} className={styles.link__a} onClick={closeNav}>
        {title}
      </a>
    </li>
  );
};

export default NavLink;
