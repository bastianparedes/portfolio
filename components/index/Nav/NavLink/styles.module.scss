@import 'styles/_breakpoints';

.link {
  &__a {
    color: rgb(255, 255, 255);
    font-size: inherit;
    text-decoration: none;

    &:hover {
      color: rgb(103, 255, 124);
    }
  }
}

@include phone-and-tablet {
  .link {
    width: 100%;
    height: 4rem;
    display: flex;
    align-items: center;
    justify-content: center;
  }
}
