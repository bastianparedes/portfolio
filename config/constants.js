const constants = {
  ABOUT: {
    CV: {
      LINK: '/files/cv.pdf',
      MESSAGE: 'Ver currículum'
    },
    DESCRIPTION_PART_1: 'Soy Bastián Paredes,',
    DESCRIPTION_PART_2:
      'Un Front End Developer con experiencia construyendo sitios web con responsive design que conecten front-end y back-end. Me especializo en el uso de JavaScript, React.js y Next.js.',
    DESCRIPTION_PART_3:
      'Siempre busco mejorar mis habilidades desarrollando proyectos propios a la par de mi trabajo. Estoy constantemente estudiando acerca de las nuevas tecnologías para actualizar mi repertorio de herramientas para que los códigos de mis creaciones sean fáciles de leer, seguros y eficientes.',
    SECTION: {
      TITLE: 'About me'
    },
    SKILLS: {
      LIST: [
        {
          NAME: 'HTML 5',
          SRC: 'html5.png'
        },
        {
          NAME: 'CSS 3',
          SRC: 'css3.png'
        },
        {
          NAME: 'JavaScript',
          SRC: 'javascript.png'
        },
        {
          NAME: 'TypeScript',
          SRC: 'typescript.png'
        },
        {
          NAME: 'React.js',
          SRC: 'react.js.png'
        },
        {
          NAME: 'Next.js',
          SRC: 'next.js.png'
        },
        {
          NAME: 'Python 3',
          SRC: 'python3.png'
        },
        {
          NAME: 'MySql',
          SRC: 'mysql.png'
        },
        {
          NAME: 'MongoDB',
          SRC: 'mongodb.svg'
        }
      ],
      TITLE: 'Skills'
    }
  },
  CONTACT: {
    LINKS: {
      EMAIL: '/contact',
      GITHUB: 'https://github.com/BastianParedes',
      INSTAGRAM: 'https://www.instagram.com/bastian.p.95/',
      LINKEDIN: 'https://www.linkedin.com/in/bastianparedes/'
    }
  },
  HOME: {
    JOB: 'Front End Developer',
    NAME: 'Bastián Paredes'
  },
  NAV: {
    BUTTONS: {
      ABOUT: {
        ID: 'about',
        TITLE: 'About'
      },
      HOME: {
        ID: 'home',
        TITLE: 'Home'
      },
      PORTFOLIO: {
        ID: 'portfolio',
        TITLE: 'Porftolio'
      }
    }
  },
  PORTFOLIO: {
    PROJECTS: [
      {
        DESCRIPTION:
          'A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.',
        IMAGE: 'falabella.png',
        LINK: 'https://www.falabella.com/falabella-cl/',
        NAME: 'Falabella',
        TECHNOLOGIES: [
          'HTML',
          'CSS',
          'JavaScript',
          'React.js',
          'Next.js',
          'TypeScript',
          'Dinamic Yield',
          'Adobe Experience Data Collection'
        ]
      },
      {
        DESCRIPTION:
          'Página web de venta de bowls de ensalada que incluye un carrito de compras y una pasarela de pago construida mediante el servicio Webpay plus de Transbank. Las compras quedan registradas en una base de datos para consultar su estado.',
        GITHUB: 'https://github.com/BastianParedes/tusalad/',
        IMAGE: 'tu-salad.png',
        LINK: '/tusalad',
        NAME: 'Tu Salad',
        TECHNOLOGIES: ['React.js', 'Next.js', 'TypeScript', 'WebPay']
      },
      {
        DESCRIPTION:
          'Es una página web que recibe las imágenes cargadas por el usuario, las configura a gusto y finalmente genera un archivo PDF que se guarda en el computador del mismo. Todo el proceso ocurre del lado del Front-end.',
        GITHUB: 'https://github.com/BastianParedes/pdf-constructor/',
        IMAGE: 'pdf-constructor.png',
        LINK: '/pdf-constructor',
        NAME: 'PDF Constructor',
        TECHNOLOGIES: ['React.js', 'SortableJS', 'jsPDF']
      },
      {
        DESCRIPTION:
          'Aplicación web que pide al usuario un nombre y lo lleva a una sala donde todos los conectados a la página en ese momento pueden conversar al mismo tiempo.',
        GITHUB: 'https://github.com/BastianParedes/my-global-chat/',
        IMAGE: 'my-global-chat.png',
        LINK: '/my-global-chat',
        NAME: 'My Global Chat',
        TECHNOLOGIES: ['React.js', 'Socket.IO']
      },
      {
        DESCRIPTION:
          'CRUD que administra las contraseñas ingresadas por el ususario, las encripta y la muestra en el Front-End.',
        GITHUB: 'https://github.com/BastianParedes/passwords/',
        IMAGE: 'passwords.png',
        LINK: '/passwords',
        NAME: 'Passwords',
        TECHNOLOGIES: ['Next.js', 'MySQL', 'Prisma', 'Bcrypt']
      },
      {
        DESCRIPTION:
          'Aplicación que genera el código QR correspondiente a la URL ingresada.',
        GITHUB: 'https://github.com/BastianParedes/qr-generator/',
        IMAGE: 'qr-generator.png',
        LINK: '/qr-generator',
        NAME: 'QR generator',
        TECHNOLOGIES: ['QRcode', 'Next.js']
      },
      {
        DESCRIPTION:
          'Aplicación que muestra el clima actual en la ubicación ingresada.',
        GITHUB: 'https://github.com/BastianParedes/weather/',
        LINK: '/weather',
        NAME: 'Weather',
        TECHNOLOGIES: ['Google Maps API', 'openWeather API', 'Next.js']
      },
      {
        DESCRIPTION:
          'Aplicación de escritorio que construye evaluaciones de seleción única de matemática para el nivel secundario de las escuelas chilenas.',
        GITHUB: 'https://github.com/BastianParedes/tstmaker/',
        IMAGE: 'tstmaker.png',
        NAME: 'TstMaker',
        TECHNOLOGIES: ['Python3', 'Tkinter', 'MiKTeX']
      }
    ]
  }
};

export default constants;
