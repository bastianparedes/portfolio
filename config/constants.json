{
  "HOME": {
    "NAME": "Bastián Paredes",
    "JOB": "Front End Developer"
  },
  "NAV": {
    "BUTTONS": {
      "HOME": {
        "ID": "home",
        "TITLE": "Home"
      },
      "ABOUT": {
        "ID": "about",
        "TITLE": "About"
      },
      "PORTFOLIO": {
        "ID": "portfolio",
        "TITLE": "Porftolio"
      }
    }
  },
  "ABOUT": {
    "SECTION": {
      "TITLE": "About me"
    },
    "DESCRIPTION_PART_1": "Soy Bastián Paredes,",
    "DESCRIPTION_PART_2": "Un Front End Developer con experiencia construyendo sitios web con responsive design que conecten front-end y back-end. Me especializo en el uso de JavaScript, React.js y Next.js.",
    "DESCRIPTION_PART_3": "Siempre busco mejorar mis habilidades desarrollando proyectos propios a la par de mi trabajo. Estoy constantemente estudiando acerca de las nuevas tecnologías para actualizar mi repertorio de herramientas para que los códigos de mis creaciones sean fáciles de leer, seguros y eficientes.",
    "CV": {
      "MESSAGE": "Ver currículum",
      "LINK": "/api/cv"
    },
    "SKILLS": {
      "TITLE": "Skills",
      "LIST": [
        {
          "SRC": "html5.png",
          "NAME": "HTML 5"
        },
        {
          "SRC": "css3.png",
          "NAME": "CSS 3"
        },
        {
          "SRC": "javascript.png",
          "NAME": "JavaScript"
        },
        {
          "SRC": "typescript.png",
          "NAME": "TypeScript"
        },
        {
          "SRC": "react.js.png",
          "NAME": "React.js"
        },
        {
          "SRC": "next.js.png",
          "NAME": "Next.js"
        },
        {
          "SRC": "python3.png",
          "NAME": "Python 3"
        },
        {
          "SRC": "mysql.png",
          "NAME": "MySql"
        },
        {
          "SRC": "mongodb.svg",
          "NAME": "MongoDB"
        }
      ]
    }
  },
  "PORTFOLIO": {
    "PROJECTS": [
      {
        "NAME": "Falabella",
        "LINK": "https://www.falabella.com/falabella-cl/",
        "IMAGE": "falabella.png",
        "GITHUB": "",
        "DESCRIPTION": "A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.",
        "TECHNOLOGIES": [
          "HTML",
          "CSS",
          "JavaScript",
          "React.js",
          "Next.js",
          "TypeScript",
          "Dinamic Yield",
          "Adobe Experience Data Collection"
        ]
      },
      {
        "NAME": "Tu Salad",
        "LINK": "https://tusalad.vercel.app/",
        "IMAGE": "tu-salad.png",
        "GITHUB": "https://github.com/BastianParedes/tusalad/",
        "DESCRIPTION": "Página web de venta de bowls de ensalada que incluye un carrito de compras y una pasarela de pago construida mediante el servicio Webpay plus de Transbank. Las compras quedan registradas en una base de datos para consultar su estado.",
        "TECHNOLOGIES": ["React.js", "Next.js", "TypeScript", "WebPay"]
      },
      {
        "NAME": "PDF Constructor",
        "LINK": "https://pdf-constructor.vercel.app/",
        "IMAGE": "pdf-constructor.png",
        "GITHUB": "https://github.com/BastianParedes/pdf-constructor/",
        "DESCRIPTION": "Es una página web que recibe las imágenes cargadas por el usuario, las configura a gusto y finalmente genera un archivo PDF que se guarda en el computador del mismo. Todo el proceso ocurre del lado del Front-end.",
        "TECHNOLOGIES": ["React.js", "SortableJS", "jsPDF"]
      },
      {
        "NAME": "My Global Chat",
        "LINK": "https://my-global-chat.vercel.app/",
        "IMAGE": "my-global-chat.png",
        "GITHUB": "https://github.com/BastianParedes/my-global-chat/",
        "DESCRIPTION": "Aplicación web que pide al usuario un nombre y lo lleva a una sala donde todos los conectados a la página en ese momento pueden conversar al mismo tiempo.",
        "TECHNOLOGIES": ["React.js", "Socket.IO"]
      },
      {
        "NAME": "TstMaker",
        "LINK": "",
        "IMAGE": "tstmaker.png",
        "GITHUB": "https://github.com/BastianParedes/tstmaker/",
        "DESCRIPTION": "Aplicación de escritorio que construye evaluaciones de seleción única de matemática para el nivel secundario de las escuelas chilenas.",
        "TECHNOLOGIES": ["Python3", "Tkinter", "MiKTeX"]
      }
    ]
  },
  "CONTACT": {
    "LINKS": {
      "GITHUB": "https://github.com/BastianParedes",
      "EMAIL": "/contact",
      "LINKEDIN": "https://www.linkedin.com/in/bastianparedes/",
      "INSTAGRAM": "https://www.instagram.com/bastian.p.95/"
    }
  },
  "FOOTER": {
    "MESSAGE": "Portfolio designed by @Bastián Paredes, © 2022"
  }
}
