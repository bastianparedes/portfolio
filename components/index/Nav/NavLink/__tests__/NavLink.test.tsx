import React from 'react'
import NavLink from '..'
import { render, screen, fireEvent } from '@testing-library/react'
import constants from '../../../../../config/constants.json'

describe('<NavLink />', () => {
  const props = {
    link: constants.NAV.BUTTONS.HOME.ID,
    title: constants.NAV.BUTTONS.HOME.TITLE,
    onClick: jest.fn()
  }

  it('should render', () => {
    const { container } = render(<NavLink {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('should open nav and closses when link clicked', () => {
    render(<NavLink {...props} />)
    screen.getByText(constants.NAV.BUTTONS.HOME.TITLE)
    fireEvent.click(screen.getByText(constants.NAV.BUTTONS.HOME.TITLE))
    expect(props.onClick).toBeCalledTimes(1)
  })
})
