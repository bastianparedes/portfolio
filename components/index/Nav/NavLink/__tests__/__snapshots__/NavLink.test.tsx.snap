// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> should render 1`] = `
<div>
  <li
    class="link"
  >
    <a
      class="link__a"
      href="#home"
    >
      Home
    </a>
  </li>
</div>
`;
